### DISCUSSION TITLE
How to use capsule body?

### DISCUSSION THREAD

**Discussion by hpx7** [enable3d/enable3d#341] [Help / Questions]
I can do `physics.add.box()` and `physics.add.sphere()`, but what's the best way to add a capsule shape?

**Comment by yandeu** [comment]
Does `physics.add.capsule()` not work?

**Reply by hpx7** [reply]
Nope, these are the ones that are available: https://github.com/enable3d/enable3d/blob/7cce6559a6bd7c39269dd4556011c0335a9f78fd/packages/ammoPhysics/src/physics.ts#L575-L609

**Reply by yandeu** [reply]
Ooh. I guess I forgot to add it :/

**Reply by yandeu** [reply]
I have added it in https://github.com/enable3d/enable3d/commit/0596f5c734232c84e39571c8786f6546eb1b8d5b.

**Reply by hpx7** [reply]
Thank you! Could you cut a release with it as well?

**Reply by yandeu** [reply]
I don't want to release a new version because of such a small change. Please for now use a function similar to the one below:

```ts
const addCapsule = (scene: Scene3D, x: number, y: number, z: number) => {
  const geometry = new THREE.CapsuleGeometry(1, 1, 4, 8)
  const material = new THREE.MeshBasicMaterial({ color: 0x00ff00, transparent: true, opacity: 0.5 })
  const capsule = new ExtendedMesh(geometry, material)
  capsule.shape = 'capsule'
  const obj = new ExtendedObject3D()
  obj.add(capsule)
  obj.position.set(x, y, z)
  scene.add.existing(obj)
  scene.physics.add.existing(obj)
  return obj
}

class MainScene extends Scene3D {
  async create() {
    this.warpSpeed()
    this.physics.debug?.enable()

    addCapsule(this, 0, 2, 0)
  }
}

PhysicsLoader('/lib', () => new Project({ scenes: [MainScene] }))
```
